### PR TITLE
[Testcase] Improve Coverage of course exercise details

### PR DESCRIPTION
**Pull request by derLalla**
### Checklist
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)

### Motivation and Context

We want to increase the overall test coverage to ~80%

### Description

This increases the test coverage of course-exercise-details.component.ts to 79,8%.

### Test Coverage

course-exercise-details.component.ts to 79,8%.

### CODE REVIEW

File: src/test/javascript/spec/component/overview/exercise-details/course-exercise-details.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+import { ParticipationWebsocketService } from 'app/overview/participation-websocket.service';
+import { ParticipationService } from 'app/exercises/shared/participation/participation.service';
+import { SourceTreeService } from 'app/exercises/programming/shared/service/sourceTree.service';
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+import { GuidedTourService } from 'app/guided-tour/guided-tour.service';
+import { CourseExerciseSubmissionResultSimulationService } from 'app/course/manage/course-exercise-submission-result-simulation.service';
+import { ProgrammingExerciseSimulationService } from 'app/exercises/programming/manage/services/programming-exercise-simulation.service';
+import { TeamService } from 'app/exercises/shared/team/team.service';
+import { QuizExerciseService } from 'app/exercises/quiz/manage/quiz-exercise.service';
+import { BehaviorSubject, empty, of } from 'rxjs';
```
Comment (nicolasruscher) [inline src/test/javascript/spec/component/overview/exercise-details/course-exercise-details.component.spec.ts:23]: the unused import can be removed

**Review by nicolasruscher — COMMENTED** [review]
apart from the unused import it looks good to me!

**Review by FuchsDominik — DISMISSED** [review]
The initialization of the stubs at the beginning bring a bit overhead since it seems they are not used in every test, but since they are used in **most** tests i think thats fine :)

**Review by nicolasruscher — APPROVED** [review]
Approving 👍

File: src/test/javascript/spec/component/overview/exercise-details/course-exercise-details.component.spec.ts (hunk 1 above)
Comment (krusche) [inline src/test/javascript/spec/component/overview/exercise-details/course-exercise-details.component.spec.ts:23]:
```suggestion
import { BehaviorSubject, of } from 'rxjs';
```

**Review by krusche — APPROVED** [review]

**Pull request merged**